From 36d3b32cc9425c51c3ae766c469590f9fe164218 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Mon, 4 Oct 2021 20:20:35 +0300
Subject: [PATCH] media: i2c: ar0233: change ADC LUT

This changes ar0233_rev2_LUT_24_to_12 to  ar0233_rev2_LUT_20_to_12

Signed-off-by: Alexander Ronzhin <alexander.ronzhin@cogentembedded.com>
Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/media/i2c/soc_camera/ar0233_rev2.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/soc_camera/ar0233_rev2.h b/drivers/media/i2c/soc_camera/ar0233_rev2.h
index e914d2b..0ac82d2 100644
--- a/drivers/media/i2c/soc_camera/ar0233_rev2.h
+++ b/drivers/media/i2c/soc_camera/ar0233_rev2.h
@@ -3579,8 +3579,8 @@ static const struct ar0xxx_reg ar0233_rev2_mipi_124p5_12bit_4lane_v21[] = {
 { }
 }; /* mipi_124p5_12bit_4lane_v21 */
 
-static const struct ar0xxx_reg ar0233_rev2_LUT_24_to_12[] = {
-{0x31AC, 0x180C},
+static const struct ar0xxx_reg ar0233_rev2_LUT_20_to_12[] = {
+{0x31AC, 0x140C},
 {0x31D0, 0x01}, //companding
 {0x33DA, 0},
 {0x33C0, 0x2000}, //LUT_00
@@ -3600,7 +3600,7 @@ static const struct ar0xxx_reg ar0233_rev2_LUT_24_to_12[] = {
 {0x33F8, 0xFF70},
 {0x33FA, 0xFF70}, //LUT_15
 { }
-}; /* LUT_24_to_12 */
+}; /* LUT_20_to_12 */
 
 static const struct ar0xxx_reg ar0233_rev2_Full_resolution[] = {
 {0x3004, AR0233_X_START}, // X_ADDR_START_
@@ -3664,7 +3664,7 @@ static const struct ar0xxx_reg *ar0233_regs_hdr_mipi_12bit_30fps_rev2[] = {
 	ar0233_rev2_HDR_3exp_12bit,
 	ar0233_rev2_pll_23_102_4lane_12b,
 	ar0233_rev2_mipi_12bit_4lane,
-//	ar0233_rev2_LUT_24_to_12,
+	ar0233_rev2_LUT_20_to_12,
 	ar0233_rev2_Full_resolution,
 	ar0233_rev2_HDR_ratio_gain_default,
 	NULL,
@@ -3679,7 +3679,7 @@ static const struct ar0xxx_reg *ar0233_regs_seplus_mipi_12bit_30fps_rev2[] = {
 	ar0233_rev2_O1_Recommended_Defaults_SE_T1_LIN_T2,
 	ar0233_rev2_disable_embed_data_stat,
 	ar0233_rev2_SEPLUS_12bit,
-//	ar0233_rev2_LUT_24_to_12,
+	ar0233_rev2_LUT_20_to_12,
 	ar0233_rev2_SEPLUS_ratio_gain_default,
 	NULL,
 };
@@ -3697,7 +3697,7 @@ static const struct ar0xxx_reg *ar0233_regs_seplus_mipi_12bit_60fps_rev2[] = {
 	ar0233_rev2_SE_Pixel_char_Recommended_REV2_V3, // If R0x300E version is earlier than 0x202D, it should be load
 	ar0233_rev2_sequence_hidy_ar0233_REV2_SE_V13, // If R0x300E version is earlier than 0x202D, it should be load
 	ar0233_rev2_SE_Lin_T2_12bit, // ar0233_rev2_SEPLUS_12bit
-//	ar0233_rev2_LUT_24_to_12,
+	ar0233_rev2_LUT_20_to_12,
 	ar0233_rev2_SE_AGain_1x, // ar0233_rev2_SEPLUS_ratio_gain_default,
 	ar0233_rev2_Tempsensor_init,
 	ar0233_rev2_SE_Design_Recommended2,
-- 
2.7.4

